packages/notebook/src/panel.ts

```diff
@@ -75,6 +80,22 @@ export class NotebookPanel extends DocumentWidget<Notebook, INotebookModel> {
     });
   }
 
+  _onSave(sender: DocumentRegistry.Context, state: DocumentRegistry.SaveState) {
+    if (state === 'started') {
+      // Find markdown cells
+      const { cells } = this.model;
+      each(cells, cell => {
+        if (isMarkdownCellModel(cell)) {
+          for (let key of cell.attachments.keys) {
+            if (!cell.value.text.includes(key)) {
```

Comment:
This is not very robust (e.g. compared to the logic in classic notebook), but I am happy to merge as-is, and post this as a follow-up issue.

Reply:
Given that this PR has been running on for a bit, I'd be happy to move this to a follow up issue too :)

Reply:
Yeah, that was the thought 👍